spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfiguration.java

```diff
@@ -0,0 +1,123 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.sql.autoconfig;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+import javax.sql.DataSource;
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.api.services.sqladmin.SQLAdmin;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.gcp.core.GcpProperties;
+import org.springframework.cloud.gcp.core.autoconfig.GcpContextAutoConfiguration;
+import org.springframework.cloud.gcp.sql.CloudSqlJdbcUrlProvider;
+import org.springframework.cloud.gcp.sql.GcpCloudSqlProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.StringUtils;
+
+/**
+ * Google Cloud SQL starter.
+ *
+ * <p>It simplifies database configuration in that only an instance and database name are provided,
+ * instead of a JDBC URL. If the instance region isn't specified, this starter attempts to get it
+ * through the Cloud SQL API.
+ *
+ * @author João André Martins
+ */
+@Configuration
+@EnableConfigurationProperties(GcpCloudSqlProperties.class)
+@ConditionalOnClass({GcpProperties.class, SQLAdmin.class})
+@AutoConfigureBefore(DataSourceAutoConfiguration.class)
+@AutoConfigureAfter(GcpContextAutoConfiguration.class)
+public class GcpCloudSqlAutoConfiguration {
+
+	@Autowired
+	private GcpCloudSqlProperties gcpCloudSqlProperties;
+
+	@Autowired
+	private GcpProperties gcpProperties;
+
+	@Bean
+	@ConditionalOnMissingBean
+	protected SQLAdmin sqlAdminService(CredentialsProvider credentialsProvider)
+			throws GeneralSecurityException, IOException {
+		HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+		JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+
+		return new SQLAdmin.Builder(httpTransport, jsonFactory,
+				new HttpCredentialsAdapter(credentialsProvider.getCredentials()))
+				.setApplicationName("Google-SQLAdminSample/0.1")
```

Comment:
Oh wow, can't believe this slipped... I never read this correctly and always thought it was a required control field. I removed it and it works fine.

Although this begs the question -- like we do for Pub/Sub, we should also be getting metrics in the Storage and Cloud SQL side. I think this deserves a new issue.